coriolis/tests/osmorphing/netpreserver/test_netplan.py

```diff
@@ -0,0 +1,99 @@
+                mock.sentinel.conn,
+                mock.sentinel.os_root_dir,
+                mock.sentinel.os_root_dir,
+                mock.sentinel.hypervisor,
+                self.event_manager,
+                self.detected_os_info,
+                mock.sentinel.osmorphing_parameters,
+                mock.sentinel.operation_timeout
+            )
+        )
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_check_net_preserver_true(self, mock_test_path, mock_list_dir):
+        mock_test_path.return_value = True
+        mock_list_dir.return_value = ["01-netplan.yaml", "notconfig.txt",
+                                      "02-other.yml"]
+
+        result = self.netpreserver.check_net_preserver()
+
+        mock_test_path.assert_called_once_with(self.netpreserver.netplan_base)
+        mock_list_dir.assert_called_once_with(self.netpreserver.netplan_base)
+        self.assertTrue(result)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_check_net_preserver_false_due_to_non_yaml(self, mock_test_path,
+                                                       mock_list_dir):
+        mock_test_path.return_value = True
+        mock_list_dir.return_value = ["config.txt", "README.md"]
+
+        result = self.netpreserver.check_net_preserver()
+
+        self.assertFalse(result)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    def test_parse_network(self, mock_list_dir, mock_read_file_sudo):
+        mock_list_dir.return_value = ["01-netplan.yaml"]
+        yaml_content = """
+            network:
+            ethernets:
+                eth0:
+                    match:
+                        macaddress: "00:11:22:33:44:55"
+                    addresses:
+                        - "192.168.1.10/24"
+                eth1:
+                    addresses:
+                        - "192.168.1.20/24"
+                        - "192.168.1.21/24"
+                eth2:
+                    match:
+                        macaddress: "AA:BB:CC:DD:EE:FF"
+        """
+        mock_read_file_sudo.return_value = yaml_content.lstrip().encode()
+
+        self.netpreserver.parse_network()
+
+        expected_info = {
+            "eth0": {
+                "mac_address": "00:11:22:33:44:55",
+                "ip_addresses": ["192.168.1.10"]
+            },
+            "eth1": {
+                "mac_address": "",
+                "ip_addresses": ["192.168.1.20", "192.168.1.21"]
+            },
+            "eth2": {
+                "mac_address": "AA:BB:CC:DD:EE:FF",
+                "ip_addresses": []
+            }
+        }
+
+        self.assertEqual(self.netpreserver.interface_info, expected_info)
```

Comment:
Please cover the cases where the IP address is a yaml object, an empty object, and also cover the fact that the yaml file could have an invalid format. (wrong indentation or syntax)

Reply:
Regarding the IP address being an object, I did not find many examples, used the example presented here:
https://people.ubuntu.com/~slyon/netplan-docs/netplan-yaml/